R/graded.R

```diff
@@ -1,27 +1,123 @@
-#' Graded object for submission value
+#' Signal a final grade for a student's submission
 #'
-#' The return value from `graded` should be returned by every
-#' `*-check` chunk when used with [grade_learnr()].
+#' `graded()` is used to signal a final grade for a submission. In general, it
+#' is most helpful when used in [grade_this()], but it is used internally by
+#' other grading functions to signal that the student's submission has been
+#' graded. If you're using [grade_this()], you'll most likely want to use the
+#' helper functions `pass()`, `fail()`, `pass_if_equal()`, and
+#' `fail_if_equal()`, rather than to call `graded()` directly.
 #'
-#' `graded()` objects are signaled to the calling functions.
+#' @section Usage in `grade_learnr()`: 
+#'   
+#'   If \pkg{gradethis} is used in a [learnr::tutorial()] with the default
+#'   [gradethis_setup()], [grade_learnr()] expects the `*-check` chunk for an
+#'   exercise to return a function. When the exercise submission is to be
+#'   graded, [grade_learnr()] will call the checking function, providing it with
+#'   a consistent exercise submission environment — see [mock_this_exercise()]
+#'   for examples of this environment. The goal of this function is to evaluate
+#'   the submission and to return a final grade via `graded()`.
 #'
-#'   * [grade_result()] ignores when grades are created. `graded()` objects must be returned
-#'   * [grade_code()] handles `graded()` objects internally
-#'   * [grade_this()] will stop execution once a `pass()`, `pass_if_equal()`,
-#'     `fail()`, `fail_if_equal()`, or `grade()` is called. To generate programmatic grades,
-#'     use `graded()` or if statements around `pass*()` and `fail*()`
+#'   In general, tutorial authors will primarily use `graded()` and its helper
+#'   functions only when using [grade_this()]. Whenever one of these functions
+#'   is called inside [grade_this()], the submission checking will stop
+#'   immediately and the appropriate grade and feedback will be returned.
+#'
+#'   Internally, [grade_this_code()], [grade_result()], and [grade_code()] all
+#'   create `graded()` objects, but each provides a different mechanism for
+#'   grading the submission and tutorial authors do not need to call `graded()`
+#'   or its helper functions within those grading functions.
+#'   
+#' @examples
+#' # Suppose our exercise asks the student to prepare and execute code that
+#' # returns the value `42`. We'll use `grade_this()` to check their
+#' # submission.
+#' #
+#' # Because we are demonstrating these functions inside R documentation, we'll
+#' # save the function returned by `grade_this()` as `this_grader()`. Calling
+#' # `this_grader()` on a mock exercise submission is equivalent to running the
+#' # check code when the student clicks "Submit Answer" in a learnr tutorial.
+#' 
+#' this_grader <- 
+#' # ```{r *-check}
```

Comment:
Don't want authors to be misled that they can use `*-check` as an actual label
```suggestion
#' # ```{r example-label-check}
```